### PR TITLE
Migrate Lagrangian-MGN example to PyTorch Geometric

### PR DESCRIPTION
**Pull request by Alexey-Kamenev**
# PhysicsNeMo Pull Request

## Description

Migrate Lagrangian-MGN example to PyTorch Geometric

## Key Changes

1. Added PyG-based Lagrangian-MGN example.
2. Renamed existing DGL-based example to `lagrangian_mgn_dgl`
3. Updated existing unit tests, added equivalency test.

### Loss

blue: DGL
red: PyG

image

### Inference:

|            |            |
|----------|----------|
| DGL | animation_0 |
| PyG  | animation_0  |

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

### CODE REVIEW

File: examples/cfd/lagrangian_mgn_dgl/loggers.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
```
Comment (mnabian) [inline examples/cfd/lagrangian_mgn_dgl/loggers.py:1]: I wish we could move all these to the launch folder...

**Review by mnabian — APPROVED** [review]

**Comment by mnabian** [comment]
This looks good to me! Approving
Please note that we now have Warp implementation of neighbor search than can be used instead of torch.cdist, but that can be a separate PR...

File: examples/cfd/lagrangian_mgn_dgl/loggers.py (hunk 1 above)
Comment (Alexey-Kamenev) [reply examples/cfd/lagrangian_mgn_dgl/loggers.py:1]: Yes, that's a good idea!
AFAIR, I didn’t do this initially because of some differences in how loggers are used in `launch` vs. the L-MGN version.

I think that, at the very least, `launch` should provide a better way to enable multiple loggers (WandB, TB, MLFlow, etc.) than what we currently have, e.g., explicitly instantiating each logger directly in the training script.
